liquidity_pool_router/src/lib.rs

```diff
@@ -202,20 +198,18 @@ impl LiquidityPoolRouterTrait for LiquidityPoolRouter {
         let amounts = get_deposit_amounts(desired_a, min_a, desired_b, min_b, reserves);
 
         let client_a = TokenClient::new(&e, token_a);
-        let nonce_a = client_a.nonce(&get_contract_id(&e));
         client_a.xfer_from(
             &Signature::Contract,
-            &nonce_a,
+            &BigInt::zero(&e),
```

Comment:
What's the why behind this change? I can't gleam it from looking at just the change.

Reply:
Good point. I updated the PR description.

Reply:
Related to this, maybe https://github.com/stellar/rs-soroban-sdk/issues/533 will improve the way we use nonces so this obscure detail will no longer exist.